packages/cpp-debug/src/browser/info/info-widget.tsx

```diff
@@ -0,0 +1,492 @@
+}
+
+/**
+ * Command state interface.
+ */
+interface CommandState {
+    value: string;
+}
+
+/**
+ * Widget which manage the communication between the debugger and the view.
+ */
+export class InfoWidget extends ReactWidget {
+    /**
+     * The DebugSessionManager.
+     */
+    protected debugSessionManager: DebugSessionManager;
+
+    /**
+     * The state for the select command.
+     */
+    protected state: CommandState;
+
+    /**
+     * The React node for display the command info.
+     */
+    protected divInfo: React.ReactNode;
+
+    /**
+     * The infoView link to this Widget.
+     */
+    protected infoView: InfoView;
+
+    constructor(debugSessionManager: DebugSessionManager, infoView: InfoView) {
+        super();
+        this.debugSessionManager = debugSessionManager;
+        this.state = {value: '--empty'};
+        this.infoView = infoView;
+    }
+
+    /**
+     * Get the info content.
+     */
+    protected async getInfoContents<T>(cmd: string): Promise<Array<T> | undefined> {
+        if (this.debugSessionManager === undefined) {
+            throw new Error('No active debug session.');
+        }
+        const session = this.debugSessionManager.currentSession;
+        if (session === undefined) {
+            throw new Error('No active debug session.');
+        }
+        try {
+            const result = await session.sendCustomRequest(cmd, undefined);
+            return result.body;
+        } catch (err) {
+            console.error(err.message);
+            return undefined;
+        }
+    }
+
+    /**
+     * Change the command info.
+     * @param event the event to be process.
+     */
+    protected changeCommand = (event: React.ChangeEvent<HTMLSelectElement>): void => {
+        if (this.debugSessionManager === undefined || this.debugSessionManager.currentSession === undefined) {
+            this.divInfo = this.renderErrorMessage(' No agents currently running.');
+            this.update();
+        } else {
+            this.state.value = event.target.value;
+            this.updateCommand();
+        }
+    }
+
+    /**
+     * Analyse the command and call the proper methode.
+     */
+    protected async updateCommand(): Promise<void> {
+        switch (this.state.value) {
+            case 'agents' : {
+                await this.processAgents();
+                break;
+            }
+            case 'queues' : {
+                await this.processQueues();
+                break;
+            }
+            case 'dispatches' : {
+                await this.processDispatches();
+                break;
+            }
+            case 'threads' : {
+                await this.processThreads();
+                break;
+            }
+        }
+        this.update();
+    }
+
+    /**
+     * Process the command info agents.
+     */
+    protected async processAgents(): Promise<void> {
+        const list = await this.getInfoContents<AgentContents>('cdt-gdb-adapter/Agents');
```

Comment:
I see you expect to use a custom debug adapter, maybe see if you can only make your contributions active if `DebugSession.capabilities` advertises itself as capable of handling those requests? You'd have to add such an entry to the `initialize` response from the DA too.